Deeploy/CommonExtensions/CodeTransformationPasses/IntrospectiveCodeTransformation.py

```
@@ -42,65 +42,76 @@ class IntrospectiveCodeTransformationMixIn():
+            buffer_filters = template.buffer_filters,
+            imports = template.imports,
+            future_imports = template.future_imports,
+            source_encoding = lexer.encoding,
+            generate_magic_comment = True,
+            strict_undefined = template.strict_undefined,
+            enable_loop = template.enable_loop,
+            reserved_names = template.reserved_names,
+        )
+        module = types.ModuleType(template.module_id)
+        code = compile(source, template.module_id, "exec")
+        exec(code, module.__dict__, module.__dict__)
```

Ruff (0.12.2)

Use of `exec` detected

(S102)